tools/talis/deployment.go

```diff
@@ -0,0 +1,256 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+func upCmd() *cobra.Command {
+	var rootDir string
+	var cfgPath string
+	var SSHPubKeyPath string
+	var SSHKeyName string
+	var DOAPIToken string
+
+	cmd := &cobra.Command{
+		Use:   "up",
+		Short: "Uses the config to spin up a distributed network",
+		Long:  "Initialize the Talis network with the provided configuration.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cfg, err := LoadConfig(rootDir)
+			if err != nil {
+				return fmt.Errorf("failed to load config: %w", err)
+			}
+
+			if len(cfg.Validators) == 0 {
+				return fmt.Errorf("no validators found in config")
+			}
+
+			// overwrite the config values if flags or env vars are set
+			// flag > env > config
+			cfg.SSHKeyName = resolveValue(SSHKeyName, EnvVarSSHKeyName, cfg.SSHKeyName)
+			cfg.SSHPubKeyPath = resolveValue(SSHPubKeyPath, EnvVarSSHKeyPath, cfg.SSHPubKeyPath)
+			cfg.DigitalOceanToken = resolveValue(DOAPIToken, EnvVarDigitalOceanToken, cfg.DigitalOceanToken)
+
+			client, err := NewClient(cfg)
+			if err != nil {
+				return fmt.Errorf("failed to create client: %w", err)
+			}
+
+			if err := client.Up(cmd.Context()); err != nil {
+				return fmt.Errorf("failed to spin up network: %w", err)
+			}
+
+			if err := client.cfg.Save(rootDir); err != nil {
+				return fmt.Errorf("failed to save config: %w", err)
+			}
+
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&SSHPubKeyPath, "ssh-pub-key-path", "s", "", "path to the user's SSH public key")
+	cmd.Flags().StringVarP(&rootDir, "directory", "d", ".", "root directory in which to initialize")
+	cmd.Flags().StringVarP(&cfgPath, "config", "c", "config.json", "name of the config")
+	cmd.Flags().StringVarP(&SSHKeyName, "ssh-key-name", "n", "", "name for the SSH key")
+
+	return cmd
+}
+
+func deployCmd() *cobra.Command {
+	var rootDir string
+	var cfgPath string
+	var SSHKeyPath string
+
+	cmd := &cobra.Command{
+		Use:   "deploy",
+		Short: "Uses the config to spin up a distributed network",
+		Long:  "Initialize the Talis network with the provided configuration.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			tarPath := filepath.Join(rootDir, "payload.tar.gz")
+			tarCmd := exec.Command("tar", "-czf", tarPath, "-C", rootDir, "payload")
```

Comment:
[proposition]
The payload is generally 400-500mb. For decent home internet, it takes time to upload to all servers. What do you think of switching to `xz` compression which will take 10sec more to compress but end up with > 50% compression of the payload.

```suggestion
			tarCmd := exec.Command("tar", "--xz", "--options", "xz:compression-level=7", "-cf", tarPath, "-C", rootDir, "payload")
```